tests/sentry/api/endpoints/test_organization_member_index.py

```diff
@@ -677,6 +677,48 @@ def test_referrer_param(self, mock_send_invite_email):
 
         mock_send_invite_email.assert_called_with("test_referrer")
 
+    @patch.object(OrganizationMember, "send_invite_email")
+    def test_integration_token_can_only_invite_member_role(self, mock_send_invite_email):
+        token = self.create_internal_integration_token(
+            user=self.user, org=self.organization, scopes=["member:write"]
+        )
+        err_message = "Integration tokens are restricted to inviting new members to have the member role only."
+
+        data = {"email": "jane@gmail.com", "role": "owner", "teams": [self.team.slug]}
+        response = self.get_error_response(
+            self.organization.slug,
+            **data,
+            extra_headers={"HTTP_AUTHORIZATION": f"Bearer {token.token}"},
+            status_code=400,
+        )
+        assert response.data[0] == err_message
+
+        data = {"email": "jane@gmail.com", "role": "manager", "teams": [self.team.slug]}
+        response = self.get_error_response(
+            self.organization.slug,
+            **data,
+            extra_headers={"HTTP_AUTHORIZATION": f"Bearer {token.token}"},
+            status_code=400,
+        )
+        assert response.data[0] == err_message
+
+        data = {"email": "jane@gmail.com", "role": "member", "teams": [self.team.slug]}
+        response = self.get_success_response(
+            self.organization.slug,
+            **data,
+            extra_headers={"HTTP_AUTHORIZATION": f"Bearer {token.token}"},
+            status_code=201,
+        )
+
+        om = OrganizationMember.objects.get(id=response.data["id"])
+        assert om.user_id is None
+        assert om.email == "jane@gmail.com"
+        assert om.role == "member"
+        assert list(om.teams.all()) == [self.team]
+        self.assert_org_member_mapping(org_member=om)
+
```

Comment:
The inviter_id of the organization member is the proxy user id of the internal integration, so we skip the check for that